File: 13_NoSQL/ExamsDb/Model/Term.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;

namespace ExamDbGenerator.Model
{
    record Term(int Year, [property: BsonRepresentation(BsonType.String)] TermType TermType, DateOnly Start, DateOnly End)
    {
        [BsonElement]
        public string Id => $"{Year}{TermType.ToString()[0]}";
    }

}

File: 13_NoSQL/examsdb-java/src/main/java/at/spengergasse/examsdb/model/Term.java
package at.spengergasse.examsdb.model;

import java.time.LocalDate;

import org.bson.codecs.pojo.annotations.BsonId;


public record Term(int year, TermType termType, LocalDate start, LocalDate end) {
    @BsonId
    public String id() {
        return String.format("%s%s", String.valueOf(year), termType.toString().charAt(0));
    }
}